usr/src/uts/common/os/log_sysevent.c

```diff
@@ -641,6 +642,10 @@ sysevent_add_attr(sysevent_attr_list_t **ev_attr_list, char *name,
 		error = nvlist_add_hrtime(*ev_attr_list, name,
 		    se_value->value.sv_time);
 		break;
+	case SE_DATA_TYPE_NVLIST:
```

Comment:
@asomers Take a look at `spa_history_log_sync()` for an example of how to stringify this nvlist in a human-readable and non-recursive way (for `zfs_dbgmsg()`).  IIRC, this captures all fields that are in the nvlist, which is not recursive (so you could add each of the nvlist entries as a flat array of KV pairs if you wanted).